Sources/Sentry/SentryReplayEnvelopeItemHeader.m

```diff
@@ -0,0 +1,35 @@
+#import "SentryReplayEnvelopeItemHeader.h"
+#import "SentryEnvelopeItemType.h"
+
+@implementation SentryReplayEnvelopeItemHeader
+
+- (instancetype)initWithType:(NSString *)type
+                   segmentId:(NSInteger)segmentId
+                      length:(NSUInteger)length
+{
+    if (self = [super initWithType:type length:length]) {
+        self.segmentId = segmentId;
+    }
+    return self;
+}
+
++ (instancetype)replayRecordingHeaderWithSegmentId:(NSInteger)segmentId length:(NSUInteger)length
+{
+    return [[self alloc] initWithType:SentryEnvelopeItemTypeReplayRecording
+                            segmentId:segmentId
+                               length:length];
+}
+
++ (instancetype)replayVideoHeaderWithSegmentId:(NSInteger)segmentId length:(NSUInteger)length
+{
+    return [[self alloc] initWithType:SentryEnvelopeItemTypeReplayVideo
+                            segmentId:segmentId
+                               length:length];
+}
+
+- (NSDictionary *)serialize
+{
+    return @{ @"type" : self.type, @"length" : @(self.length), @"segment_id" : @(self.segmentId) };
```

Comment:
the `segment_id` is kind of an internal header within the payload, so it shouldn't be on the envelope item header afaiu, i.e.:

```json
// item header
{"type": "replay_video", "length": 5000}
// payload
{"segment_id": 0}\n{"type": 5, "timestamp": 1234, ...}
```

Reply:
that's correct

Reply:
Yeah, I saw the new envelope format and realised that.